services/consensus_topic_info.proto

```diff
@@ -31,60 +38,121 @@ import "duration.proto";
+     * SHA-384 digest are, in order:
+     * <ol>
+     *   <li>The previous running hash of the topic (48 bytes)</li>
+     *   <li>The `topicRunningHashVersion` (8 bytes)</li>
+     *   <li>The payer account's shard (8 bytes)</li>
+     *   <li>The payer account's realm (8 bytes)</li>
+     *   <li>The payer account's number (8 bytes)</li>
+     *   <li>The topic's shard (8 bytes)</li>
+     *   <li>The topic's realm (8 bytes)</li>
+     *   <li>The topic's number (8 bytes)</li>
+     *   <li>The number of seconds since the epoch when the
+     *       `ConsensusSubmitMessage` reached consensus (8 bytes)</li>
+     *   <li>The number of nanoseconds within the second when the
+     *       `ConsensusSubmitMessage` reached consensus (4 bytes)</li>
+     *   <li>The `topicSequenceNumber` (8 bytes)</li>
+     *   <li>The output of a SHA-384 digest of the message bytes from the
+     *       `ConsensusSubmitMessage` (48 bytes)</li>
+     * </ol>
      */
     bytes runningHash = 2;
 
     /**
-     * Sequence number (starting at 1 for the first submitMessage) of messages on the topic.
+     * A current sequence number (starting at 1 for the first message)
+     * for messages on this topic.
      */
     uint64 sequenceNumber = 3;
 
     /**
-     * Effective consensus timestamp at (and after) which submitMessage calls will no longer succeed on the topic
-     * and the topic will expire and after AUTORENEW_GRACE_PERIOD be automatically deleted.
+     * An expiration time for this topic, in seconds since the epoch.
+     * <p>
+     * For this purpose, `epoch` SHALL be the UNIX epoch
+     * with 0 at `1970-01-01T00:00:00.000Z`.
      */
     Timestamp expirationTime = 4;
 
     /**
-     * Access control for update/delete of the topic. Null if there is no key.
+     * A key that MUST sign any transaction to update or delete this topic.
+     * <p>
+     * If this value is not set (null) then the topic CANNOT be deleted,
+     * modified, or updated.
      */
     Key adminKey = 5;
 
     /**
-     * Access control for ConsensusService.submitMessage. Null if there is no key.
+     * A key that MUST sign any transaction to submit a message to this topic.
+     * <p>
+     * If this value is not set (null) then any account MAY submit messages to
+     * this topic.
      */
     Key submitKey = 6;
 
     /**
-     * If an auto-renew account is specified, when the topic expires, its lifetime will be extended
-     * by up to this duration (depending on the solvency of the auto-renew account). If the
-     * auto-renew account has no funds at all, the topic will be deleted instead.
+     * A duration, in seconds, to extend the `expirationTime` value when
+     * this topic is automatically renewed.
+     * <p>
+     * If the `autoRenewAccount` value for this topic is set to a valid account
+     * with sufficient HBAR balance to pay renewal fees when this topic
+     * expires, the system SHALL automatically renew this topic, extending the
```

Comment:
Is this a description or a requirement?

Reply:
Requirement, just a complex one to express.